libvmaf/src/meson.build

```
@@ -94,6 +96,34 @@ if is_asm_enabled
     endif
 endif
 
+json_model_c_sources = []
+built_in_models_enabled = get_option('built_in_models') == true
+
+if built_in_models_enabled
+    xxd = find_program('xxd')
```

just found out that `find_program` has `required` set to true by default https://mesonbuild.com/Reference-manual.html#find_program

---

Do we need to change anything? If `required` was true shouldn't the msys2 build have failed before I added the `xxd` package to the CI build?

---

I believe we do since it seems it will automatically fail on that line if `xxd` is not found https://gitlab.com/Luigi311/encoders-docker/-/jobs/870743576#L837

/cc @luigi311

---

```suggestion
    xxd = find_program('xxd', required : false)
```
I believe the required change is

---

I can confirm that this is an issue, i added xxd to my dependencies and it built successfully.

---

Pushed aa85f6db2e2357221ac0bb339633c03e3cea9f9f, should not be an issue anymore. Let me know if it is.

---

It no longer instantly fails but it still does fail during the build process. Here is the build log https://gitlab.com/Luigi311/encoders-docker/-/jobs/870828946 i dont see any actual error messages though

---

```c
../src/model.c:36:31: error: zero or negative size array 'built_in_models'
   36 | static const VmafBuiltInModel built_in_models[] = {
      |                               ^~~~~~~~~~~~~~~
```

---

Oh, interesting. I tested this and my compiled did not mind this.

---

```none
C compiler for the host machine: cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
C++ compiler for the host machine: c++ (gcc 9.3.0 "c++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
C++ linker for the host machine: c++ ld.bfd 2.34
```
comes from the `ubuntu:20.04` container

---

This is with gcc-9-base:amd64 (9.3.0-17ubuntu1~20.04)

---

Nevermind, It's always wrong even if the compiler doesn't blow up. I will fix.

ISO 9899:2011 6.7.6.2:
> If the expression is a constant expression, it shall have a value greater than zero.

---

PR #729 should fix this. I'll merge when CI is green.

---

I am not longer getting build errors. Thanks @kylophone